### PR TITLE
Ingestor task queue backed by Redis

### PR DESCRIPTION
**Pull request by sunu**

### CODE REVIEW

**Comment by pudo** [comment]
This is looking pretty good! Some comments: 

* Should this live in a different module than the `redis` base functions? I was thinking maybe we'd want to have a `servicelayer.queue` or something like that? 
* What are we using as a trigger criterion for Aleph (or later, a third service - but I'd like to keep it simple at first) to start indexing entities from `balkhash`? At the moment, the stats are being reset the moment the process completes - but this means nothing can be used to identify a task that has finished. We need to have some sort of answer to this ...

**Comment by sunu** [comment]
> Should this live in a different module than the `redis` base functions? I was thinking maybe we'd want to have a `servicelayer.queue` or something like that?

Yeah, makes sense. I'll move the queue specific stuff out of cache module.
 
> What are we using as a trigger criterion for Aleph (or later, a third service - but I'd like to keep it simple at first) to start indexing entities from `balkhash`? At the moment, the stats are being reset the moment the process completes - but this means nothing can be used to identify a task that has finished. We need to have some sort of answer to this ...

I have 2 possible implementations in mind for this. One is to have a separate list/queue containing datasets ready to be bulk loaded. We will push dataset ids to this queue just before calling reset_status. Aleph or the bulk push service will then poll this queue.

Or we could just trigger the bulk load directly from within service layer once we are done with a dataset. The first is a bit cleaner but the second is much simpler in my opinion. What do you think?

I plan to finish the part of queuing tasks from Aleph and executing them in ingestors first; and only after that I'll get to the bulkloading from balkhash part.

File: servicelayer/settings.py [hunk 1]
```diff
@@ -33,3 +34,7 @@
     QUEUE_MEDIUM,
     QUEUE_LOW,
 ]
+
+INGESTOR_THREADS = env.to_int(
```
Comment (pudo) [inline servicelayer/settings.py]: Perhaps this should live in ingestors?

File: servicelayer/queue.py [hunk 2]
```diff
@@ -9,17 +13,24 @@
...
 
 def _serialize(data):
+    data['entity'] = data['entity'].to_dict()
     return json.dumps(data)
 
 
 def _deserialize(data):
-    return json.loads(data)
+    data = json.loads(data)
+    data['entity'] = model.get_proxy(data['entity'])
```
Comment (pudo) [inline servicelayer/queue.py]: Do we need servicelayer to depend on followthemoney? Are there other benefits to making that link?

**Pull request merged**